### PR TITLE
Added business logic into CoffeeConroller

### PR DESCRIPTION
**Pull request by DiskReader**

### CODE REVIEW

File: CoffeeShop/Controllers/CoffeeController.cs [hunk 1]
```diff
@@ -26,13 +26,33 @@ public IEnumerable<Coffee> GetList()
...
+        public void PutById(int id, [FromBody]Coffee coffee)
         {
+            if (!db.Coffees.Any(x => x.Id == coffee.Id))
+            {
+                throw new ArgumentException("Coffee with this id does not exist", nameof(coffee));
+            }
+
+            if (!IsCoffeeValid(coffee))
+            {
+                throw new ArgumentException("Entered data is not correct", nameof(coffee));
```
Comment (hromau) [inline CoffeeShop/Controllers/CoffeeController.cs]: duplicated code. let's think about that

File: CoffeeShop/Controllers/CoffeeController.cs [hunk 2]
```diff
@@ -26,13 +26,33 @@ public IEnumerable<Coffee> GetList()
         [HttpPost]
         public void PostNewCoffee([FromBody]Coffee coffee)
```
Comment (hromau) [inline CoffeeShop/Controllers/CoffeeController.cs]: post new cofee?

File: CoffeeShop/Controllers/CoffeeController.cs [hunk 3]
```diff
@@ -26,13 +26,33 @@ public IEnumerable<Coffee> GetList()
...
+                throw new ArgumentException("Entered data is not correct", nameof(coffee));
+            }
+
             db.Coffees.Add(coffee);
             db.SaveChanges();
         }
 
         [HttpPut]
-        public void PutById(int id, [FromBody] Coffee coffee)
+        public void PutById(int id, [FromBody]Coffee coffee)
```
Comment (hromau) [inline CoffeeShop/Controllers/CoffeeController.cs]: and put by id?

**Review by hromau — CHANGES_REQUESTED** [review]

File: CoffeeShop/Context/CoffeeShopContext.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 entity.Property(e => e.Id).ValueGeneratedNever();
 
                 entity.Property(e => e.Name)
-                    .HasMaxLength(10)
+                    .HasMaxLength(30)
```
Comment (hromau) [inline CoffeeShop/Context/CoffeeShopContext.cs:35]: do u create migration on this change?

**Review by hromau — CHANGES_REQUESTED** [review]

File: CoffeeShop/Context/CoffeeShopContext.cs (hunk 4 above)
Comment (DiskReader) [reply CoffeeShop/Context/CoffeeShopContext.cs:35]: yep

**Review by CrazyKamenwik — APPROVED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Review by hromau — APPROVED** [review]

**Pull request merged**